pytest_ansible/fixtures.py

```diff
@@ -1,61 +1,39 @@
-import logging
-from .errors import AnsibleHostUnreachable
-
-try:
-    from logging import NullHandler
-except ImportError:
-    from logging import Handler
+"""PyTest fixtures."""
 
-    class NullHandler(Handler):
+import pytest
+from pytest_ansible.logger import get_logger
 
-        def emit(self, record):
-            pass
+__all__ = ('ansible_module', 'ansible_facts', 'ansible_adhoc', 'localhost')
+log = get_logger(__name__)
 
-log = logging.getLogger(__name__)
-log.addHandler(NullHandler())
 
+@pytest.fixture(scope='function')
+def ansible_adhoc(request):
+    """Return an inventory initialization method."""
+    plugin = request.config.pluginmanager.getplugin("ansible")
 
-@pytest.fixture(scope='class')
-def ansible_module_cls(request):
-    '''
-    Return AnsibleV1Module instance with class scope.
-    '''
-    warnings.warn("Use of ansible_module_cls is deprecated and will be removed in a future release", DeprecationWarning)
-    ansible_helper = request.config.pluginmanager.getplugin("ansible")
-    return ansible_helper.initialize(request)
+    def init_host_mgr(**kwargs):
+        return plugin.initialize(request, **kwargs)
+    return init_host_mgr
 
 
 @pytest.fixture(scope='function')
-def ansible_module(request):
-    '''
-    Return AnsibleV1Module instance with function scope.
-    '''
-    ansible_helper = request.config.pluginmanager.getplugin("ansible")
-    return ansible_helper.initialize(request)
+def ansible_module(request, ansible_adhoc):
+    """Return a subclass of BaseModuleDispatcher."""
+    host_mgr = ansible_adhoc()
+    return getattr(host_mgr, host_mgr.options['host_pattern'])
 
 
 @pytest.fixture(scope='function')
-def ansible_facts_cls(ansible_module_cls):
-    '''
-    Return ansible_facts dictionary
-    '''
-    warnings.warn("Use of ansible_facts_cls is deprecated and will be removed in a future release", DeprecationWarning)
-    try:
-        return ansible_module_cls.setup()
-    except AnsibleHostUnreachable, e:
-        log.warning("Hosts unreachable: %s" % e.dark.keys())
-        return e.contacted
+def ansible_facts(ansible_module):
+    """Return ansible_facts dictionary."""
+    return ansible_module.setup()
 
 
 @pytest.fixture(scope='function')
-def ansible_facts(ansible_module):
-    '''
-    Return ansible_facts dictionary
-    '''
-    try:
-        return ansible_module.setup()
-    except AnsibleHostUnreachable, e:
-        log.warning("Hosts unreachable: %s" % e.dark.keys())
-        return e.contacted
+def localhost(request):
+    """Return a host manager representing localhost."""
+    # NOTE: Do not use ansible_adhoc as a dependent fixture since that will assert specific command-line parameters have
+    # been supplied.  In the case of localhost, the parameters are provided as kwargs below.
+    plugin = request.config.pluginmanager.getplugin("ansible")
+    return plugin.initialize(request, inventory='localhost,', connection='local', host_pattern='localhost').localhost
```

Comment:
I can see a scenario where they want to escalate.  Dunno if it needs to be restricted yet.  Thanks for the comment.